### ISSUE TITLE
The high resolution images lead to artifacts.

### ISSUE THREAD

**Issue by alexaex** [piddnad/DDColor#60]
Hi, @piddnad 

I built a ddcolor model using ConvNextTiny, Pixel Decoder and Color Decoder. I didn't use the official code. So I can do the experiments flexible. The only difference is I directly use the ab chrominance with normalization in the range of -1 to 1 as the output of the model. I believe this helps numerical stability and gradient stability.

**I notice two interesting facts.** 
1. **The resolution is important.**
I trained the model with Unsplash25k about 30K iterations. And it performs well on 256x256 images. But when I try to colorize images with resolution of 1024x1024, there are unbearable artifacts. I am trying to analyze this issue. But some images worked well at both resolutions.
2. **VGG16 without batch normalization might be better.**
I also did the experiments with VGG16 without batch normalization for perceptual loss. I noticed that VGG16 perceptual loss may produce more colorful images. But the model converges slower than the perceptual loss with batch normalization, and I have to adjust the weight of each loss to balance the magnitude scale. 

The images below respectively show the results for low resolution, high resolution, the difference between the ground truth and the high-resolution output and each channel's difference. I hope this helps.
![LR](https://github.com/user-attachments/assets/c0a456ea-796a-45b7-b31e-7b701bf8dda1)
![HR](https://github.com/user-attachments/assets/9a3c8437-f2a1-46fa-995c-2ee4ffca980f)
![Image](https://github.com/user-attachments/assets/c7d7c292-e4d2-468f-b2b0-45d5a65947c0)
![Image](https://github.com/user-attachments/assets/877479b5-473e-414b-9103-c786b0f80fc4)
Here is the ground truth.
![Image](https://github.com/user-attachments/assets/8eb45480-1622-4dcc-a2a5-d524ab53ecdd)

**Comment by alexaex** [comment]
> Hi, [@piddnad](https://github.com/piddnad)
> 
> I built a ddcolor model using ConvNextTiny, Pixel Decoder and Color Decoder. I didn't use the official code. So I can do the experiments flexible. The only difference is I directly use the ab chrominance with normalization in the range of -1 to 1 as the output of the model. I believe this helps numerical stability and gradient stability.
> 
> **I notice two interesting facts.**
> 
> 1. **The resolution is important.**
>    I trained the model with Unsplash25k about 30K iterations. And it performs well on 256x256 images. But when I try to colorize images with resolution of 1024x1024, there are unbearable artifacts. I am trying to analyze this issue. But some images worked well at both resolutions.
> 2. **VGG16 without batch normalization might be better.**
>    I also did the experiments with VGG16 without batch normalization for perceptual loss. I noticed that VGG16 perceptual loss may produce more colorful images. But the model converges slower than the perceptual loss with batch normalization, and I have to adjust the weight of each loss to balance the magnitude scale.
> 
> The images below respectively show the results for low resolution, high resolution, the difference between the ground truth and the high-resolution output and each channel's difference. I hope this helps. ![LR](https://github.com/user-attachments/assets/c0a456ea-796a-45b7-b31e-7b701bf8dda1) ![HR](https://github.com/user-attachments/assets/9a3c8437-f2a1-46fa-995c-2ee4ffca980f) ![Image](https://github.com/user-attachments/assets/c7d7c292-e4d2-468f-b2b0-45d5a65947c0) ![Image](https://github.com/user-attachments/assets/877479b5-473e-414b-9103-c786b0f80fc4) Here is the ground truth. ![Image](https://github.com/user-attachments/assets/8eb45480-1622-4dcc-a2a5-d524ab53ecdd)

After carefully reviewing the inference pipeline, I supposed that the model may not predict chrominance accurately at high resolutions. Fortunately, our eyes are more sensitive to luminance than chrominance. Interpolating the low resolution chrominance to high resolution and then concatenating it with the high-resolution luminance, it works well. So I think this is a trade-off. Nice work!
